app/ota_client.py

```diff
@@ -237,98 +242,106 @@ def download(
         return error_count
 
 
+def de_escape(s: str) -> str:
+    return s.replace(r"'\''", r"'")
+
+
+@dataclass
```

Comment:
I think dataclass decorator is not required since __init__ function is implemented.
The dataclass with __init__ function seems to be complicated for me.
https://stackoverflow.com/questions/54927763/control-initialize-order-when-python-dataclass-inheriting-a-class
I'd like to revert these changes.

Reply:
Oh I don't use the dataclass generated __init__ method, but write one, so the problem stated in your link doesn't exist in our case. Actually the behavior is the same as before, and we get the benefits from the dataclass.

Reply:
> we get the benefits from the dataclass.

In this change, what is the benefits from the dataclass?

Reply:
Currently most important is dataclass will generate `__repr__` , `__eq__` and `__hash__`  for us automatically. 
`__repr__` is for debug purpose, `__eq__` and `__hash__` will allow customize set/dict behavior when storing Inf related class' instance(this feature will be used in later development).